rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -588,22 +623,26 @@ object ExecutionBatch extends Loggable {
           , NoAnswerReportType
         )
 
-      case UnexpectedVersion(runTime, runVersion, runExpiration, expectedVersion, expectedExpiration) =>
+      case UnexpectedVersion(runTime, Some(runVersion), runExpiration, expectedVersion, expectedExpiration) =>
```

Comment:
case don't seem to be exhaustive
What happen here if the Option is none (same for next case)